tests/conftest.py

```
@@ -0,0 +1,59 @@
+    shutil.rmtree(tmpdir, ignore_errors=True)
+
+
+@pytest.fixture
+def mock_git_repo(temp_dir):
+    """Create a mock git repository for testing."""
+    repo_path = os.path.join(temp_dir, "test_repo")
+    os.makedirs(repo_path)
+
+    # Initialize a git repo
+    os.chdir(repo_path)
+    os.system("git init")
```

Ruff (0.14.10)

Starting a process with a shell: seems safe, but may be changed in the future; consider rewriting without `shell`

(S605)